session/pingpong/invoice_tracker.go

```diff
@@ -271,9 +279,25 @@ func (it *InvoiceTracker) requestPromise(r []byte, pm crypto.ExchangeMessage) er
 		it.updateFee()
 	}
 
+	details := rRecoveryDetails{
+		R:           hex.EncodeToString(r),
+		AgreementID: pm.AgreementID,
+	}
+
+	bytes, err := json.Marshal(details)
+	if err != nil {
+		return errors.Wrap(err, "could not marshal R recovery details")
```

Comment:
old wrap?

Reply:
This whole component needs to be changed, will do it in a separate pr someday.